### ISSUE TITLE
Issue `(nyxt:start :no-socket t)` twice from REPL

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2732]
Starting Nyxt twice or more from the REPL (without a socket) has funny
consequences.

Steps to reproduce:

- `(asdf:load-system "nyxt/gi-gtk")`
- `(nyxt:start :no-socket t)`
- `(nyxt:start :no-socket t)`

For instance, issuing `execute-command` in the firstly created Nyxt window opens
the prompt-buffer in the secondly created Nyxt window.

Issuing `nyxt:quit` in one of the windows makes the other one unusable.

In short, both instances are not independent as expected.

**Comment by aadcg** [comment]
The issue doesn't occur in the binary so it's probably not a priority.

**Comment by Ambrevar** [comment]
Yeah, this makes total sense, since there can be only one `*browser*` in the image!

If you want multiple browsers, you've got to instantiate a `browser` yourself and bind `*browser*` locally.  This is by design.
Is there something we can improve here?

**Comment by aadcg** [comment]
> there can be only one *browser* in the image!

Can you explain the rationale behind the design?

**Comment by Ambrevar** [comment]
`*browser*` is a convenient dynamic global.  Without it, you'd have to pass a browser arg to _all_ function calls.

**Comment by aadcg** [comment]
That is clear. But my question was: why do we limit ourselves to creating a single instance of class browser?

**Comment by Ambrevar** [comment]
I don't understand your question.  You can run `(make-instance 'browser)` as many times as you want.  Is that what you meant?

**Comment by aadcg** [comment]
Ok, let me explain the way I see it.

There are two options: either we allow the existence of multiple browser instances in a single Lisp process or not. For simplicity, I think we should have a single one. That's why we define the global variable `*browser*`.

Locally let-binding `*browser*` is a way to go around this constraint of having a single browser instance. But that's mainly to do something a bit off-the-chart like testing.

So, the issue I'm raising here is: the second call to `(nyxt:start :no-socket t)` shouldn't be valid since it gives the false expectation that you can have two independent browser initialized from the same Lisp process.

Does that make sense?

**Comment by Ambrevar** [comment]
You are right, we should guard against `*browser*` being non-nil.  I'll send a patch.

**Comment by Ambrevar** [comment]
See #2735.
